yaksh/views.py

```diff
@@ -1843,6 +1843,145 @@ def download_assignment_file(request, quiz_id, question_id=None, user_id=None):
+        required_fields = ['firstname', 'lastname', 'email']
+        try:
+            reader = csv.DictReader(csv_file.read().decode('utf-8').splitlines(),
+                                    dialect=dialect)
+        except TypeError:
+            context['message'] = "Bad CSV file"
+            return my_render_to_response('yaksh/course_detail.html', context,
+                                         context_instance=ci)
+        stripped_fieldnames = [field.strip().lower() for field in reader.fieldnames]
+        for field in required_fields:
+            if field not in stripped_fieldnames:
+                context['message'] = "The CSV file does not contain the required headers"
+                return my_render_to_response('yaksh/course_detail.html', context,
+                                             context_instance=ci)
+        reader.fieldnames = stripped_fieldnames
+        context['upload_details'] = _read_user_csv(reader, course)
+    return my_render_to_response('yaksh/course_detail.html', context,
+                                 context_instance=ci)
+
+
+def _read_user_csv(reader, course):
+    fields = reader.fieldnames
+    upload_details = ["Upload Summary:"]
+    counter = 0
+    for row in reader:
+        counter += 1
+        (username, email, first_name, last_name, password, roll_no, institute,
+         department, remove) = _get_csv_values(row, fields)
+        if not email or not first_name or not last_name:
+            upload_details.append("{0} -- Missing Values".format(counter))
+            continue
+        users = User.objects.filter(username=username)
+        if users.exists():
+            user = users[0]
+            if remove.strip().lower() == 'true':
+                if _remove_from_course(user, course):
+                    upload_details.append("{0} -- {1} -- User rejected".format(
+                                          counter, user.username))
+                    continue
+            else:
+                if _add_to_course(user, course):
+                    upload_details.append("{0} -- {1} -- User rejected".format(
+                                          counter, user.username))
+            if user not in course.get_enrolled():
+                upload_details.append("{0} -- {1} not added to course".format(
+                    counter, user))
+                continue
+        user_defaults = {'email': email, 'first_name': first_name,
+                         'last_name': last_name}
+        user, created = _create_or_update_user(username, user_defaults)
+        profile_defaults = {'institute': institute, 'roll_number': roll_no,
+                            'department': department, 'is_email_verified': True}
+        _create_or_update_profile(user, profile_defaults)
+        if created:
+            state = "Added"
+            user.set_password(password)
+            user.save()
+            course.students.add(user)
+        else:
+            state = "Updated"
```

Comment:
while updating if I change password field for a user the password is not updated. I guess you need to add below lines outside if and else 
<pre>user.set_password(password) 
user.save()</pre>

Reply:
Yes, initially thought should not allow update password by the moderator.
But, yes this is required, incase the user changes the password and forget, then moderator can update!